spectrum-offchain-lm/src/executor.rs

```diff
@@ -218,14 +218,11 @@ where
                                             for i in invalidations {
                                                 match i {
                                                     Invalidation::Pool => {
-                                                        self.pool_repo
-                                                            .lock()
-                                                            .await
-                                                            .invalidate(
-                                                                pool.get_self_state_ref(),
-                                                                pool.get_self_ref(),
-                                                            )
-                                                            .await;
+                                                        // Here the TX is rejected by the node
+                                                        // because of an invalid pool box. We
+                                                        // should return the order to the backlog
+                                                        // and wait for updated pool box.
+                                                        self.backlog.lock().await.suspend(ord.clone()).await;
```

Comment:
This doesn't guarantee this invalid pool won't be used with the subsequent orders before new pool arrives. invalidation + order suspension + ledger update loop should be used in conjunction.

Reply:
Oh yeah, of course. I forgot that `ConfirmedProgramUpdateHandler` can 'resume' an existing program.